yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -233,6 +229,15 @@ private[spark] class Client(
           "for alternatives.")
     }
 
+    def addDistributedUri(uri: URI): Unit = {
+      val uriStr = uri.toString()
+      if (distributedUris.contains(uriStr)) {
+        throw new IllegalArgumentException(
```

Comment:
Do we need to throw an exception here?  What's the harm in just logging a warning and not adding the URI?